synapse/handlers/receipts.py

```diff
@@ -118,7 +118,7 @@ def received_client_receipt(self, room_id, receipt_type, user_id,
         if not is_new:
             return
 
-        self.federation.send_read_receipt(receipt)
+        yield self.federation.send_read_receipt(receipt)
```

Comment:
This is technically unrelated, but I just spotted it and it's near enough. #4852 introduced a logcontext leak here.